rfcs/text/0000-planning-for-the-dcp.md

```diff
@@ -0,0 +1,211 @@
+### DCP PR:
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# Planning for the DCP
+
+## Summary
+
+This RFC adopts and formalizes existing best project management practices and applications from the DCP community to define one process for planning and delivering product increments of the DCP *data as a service* on a regular cadence.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED" "MAY", and "OPTIONAL" in this document are to be interpreted as described in BCP 14, [RFC2119](https://www.rfc-editor.org/rfc/rfc2119.txt), and [RFC8174](https://www.rfc-editor.org/rfc/rfc8174.txt) when, and only when, they appear in all capitals, as shown here.
+
+## Author(s)
+
+ [Trevor Heathorn](mailto:theathor@ucsc.edu)
+ 
+ [Brian Raymor](mailto:brianraymor@chanzuckerberg.com)
+
+## Shepherd
+
+[TBD](mailto:username@example.com)
+
+## Motivation
+
+***[Note - remove if approved]***
+*Early versions of this RFC were socialized with the DCP community during a PM call and at the DCP meeting in February 2019* 
+
+One of the **challenges** facing the DCP as it matures was identified in this [Update from HCA Science Governance](https://docs.google.com/presentation/d/13hZ7vmr371mN1YFjFyDN-6c0Ul95NwiJcVsjA3nJFVE/edit#slide=id.p3):
+
+> "A lack of reproducible planning process that prioritizes deliverables and ensures confidence that they will meet the needs of HCA"
```

Comment:
"reproducible" means that we can _turn the crank_ on a regular cadence - that there is a repeatable planning/execution process that enables DCP to generate quarterly releases with high-confidence based on a prioritized roadmap. One of the **Immediate Goals** recently communicated from Oversight to DCP PM:

>Establish a quarterly release cycle (both data and software) and align efforts across DCP development

Reply:
@diekhans - **Tiger teams** (_balancing developer resources across boxes_) is outside the scope of this RFC. Our inability to move resources between components/institutions based on DCP priorities has been communicated to some members of Oversight as impacting our agility. 

Your conclusion:

>Perhaps because I am not a PM, I am not seeing how this proposal helps to accelerate the delivery of features

Within the current proposal, this is addressed by:
1. Prioritizing DCP features. If AUDR is Pri 0, then _all hands on deck_! Currently, everything is equally important which means that community members may be unsure whether to focus on Feature A or Feature B or Feature C. They may end up poorly time slicing between both. (And the nascent roadmap process should guide prioritization in the future)
2. Assigning a shepherd to each DCP Feature to help align component teams and drive progress.

**4 Week Sprints** - Our motivations:

1. Scrum _ceremonies_ require valuable meeting time (within a small time window) on a regular cadence which is especially difficult to schedule for our geographically distributed teams. We'd prefer to pay that cost every four weeks instead of every two weeks. 
2. We want to encourage the use of _Spikes_ to thoughtfully explore/prototype complex problems and avoid your scenario where component Foo discovers an unknown dependency on component Bar in the middle of a Sprint. But Sprints can also be cancelled per the Scrum process. For example, a more realistic scenario might be the Metadata user experience during Beta. If we had started a Sprint and received this feedback, then we may have cancelled the Sprint and refocused our Sprint Goal.

And there's a great deal of literature on the **best** duration of Sprints.

Reply:
@brianraymor, can you go into more detail on the process around cancelled sprints? 
My concern is if we cancelled a sprint each time we realized we need to add in a ticket, it would be very difficult to get through a complete 4-week sprint. 

In the scenario you mentioned above, could component Bar replace an item in their sprint to make room for the urgent dependency?

Reply:
@samanehsan - Please see **Cancelling a Sprint** in the [Scrum Guide](https://scrumguides.org/docs/scrumguide/v2017/2017-Scrum-Guide-US.pdf). It's an extreme and unusual action. I believe that we would have cancelled a Sprint to respond to the Metadata feedback during Beta for example. 

> My concern is if we cancelled a sprint each time we realized we need to add in a ticket

You can **add** tickets per Scrum:

*As new work is required, the Development Team adds it to the Sprint Backlog. As work is
performed or completed, the estimated remaining work is updated. When elements of the plan
are deemed unnecessary, they are removed. Only the Development Team can change its Sprint
Backlog during a Sprint.*

> In the scenario you mentioned above, could component Bar replace an item in their sprint to make room for the urgent dependency?

It's had to discuss in the abstract. Let's be concrete:

0. Sprint starts.
1. Component Foo is blocked on a change by Component Bar in the Sprint Backlog for a DCP priority.
2. Component Baz suddenly discovers an urgent dependency on Component Bar.
3. Component Bar cannot do both.
3a. If the urgent dependency is **not** related to a DCP priority, then it's declined unless Component Baz has buffer to do the work.
3b. If the urgent dependency is related to a DCP priority, then this would need to be surfaced to the Product Owners for a decision. It would also need to be discussed during the Retrospective to understand why the urgent dependency was not understood. We're encouraging Spikes for research and prototypes. 

There are [multiple strategies](https://medium.com/agilelab/strategies-for-handling-unplanned-work-during-sprint-2f89697509ff) for managing such cases that we should consider.

Reply:
re: cancelling a sprint, we are chronically short of meeting rooms at the EBI right now. Depending on the number of people who need to be involved (hence size of meeting room) it will be pretty difficult for us to shift the sprint window once established. Not saying this is a blocker to cancellations but one should bear it in mind.